core/include/usFramework.h

```diff
@@ -28,74 +28,176 @@
+     * stop of this Framework has completed. This method will only wait if
+     * called when this Framework is in the {@link #STATE_STARTING}, {@link #STATE_ACTIVE},
+     * or {@link #STATE_STOPPING} states. Otherwise it will return immediately.
+     * <p>
+     * A Framework Event is returned to indicate why this Framework has stopped.
+     *
+     * @param timeout Maximum time duration to wait until this
+     *        Framework has completely stopped. A value of zero will wait
+     *        indefinitely.
+     * @return A Framework Event indicating the reason this method returned. The
+     *         following {@code FrameworkEvent} types may be returned by this
+     *         method.
+     *         <ul>
+     *         <li>{@link FrameworkEvent#STOPPED STOPPED} - This Framework has
+     *         been stopped. </li>
+     *
+     *
+     *         <li>{@link FrameworkEvent#ERROR ERROR} - The Framework
+     *         encountered an error while shutting down or an error has occurred
+     *         which forced the framework to shutdown. </li>
+     *
+     *         <li> {@link FrameworkEvent#WAIT_TIMEDOUT WAIT_TIMEDOUT} - This
+     *         method has timed out and returned before this Framework has
+     *         stopped.</li>
+     *         </ul>
+     */
+    FrameworkEvent WaitForStop(const std::chrono::milliseconds& timeout);
 
     /**
      * Start this Framework.
      *
      * <p>
      * The following steps are taken to start this Framework:
-     * -# If this Framework is not in the STARTING state, start this Framework.
-     * -# This Framework's state is set to ACTIVE.
+     * -# If this Framework is not in the {@link #STATE_STARTING} state,
+     *    {@link #Init() initialize} this Framework.
+     * -# All installed bundles must be started in accordance with each
+     *    bundle's persistent <i>autostart setting</i>. This means some bundles
+     *    will not be started, some will be started with <i>eager activation</i>
+     *    and some will be started with their <i>declared activation</i> policy.
+     *    Any exceptions that occur during bundle starting are wrapped in a
+     *    \c std::runtime_error and then published as a framework event of type
+     *    {@link FrameworkEvent#ERROR}
+     * -# This Framework's state is set to {@link #STATE_ACTIVE}.
+     * -# A framework event of type {@link FrameworkEvent#STARTED} is fired
      *
      * @throws std::runtime_error If this Framework could not be started.
      */
     void Start();
 
     /**
+     * Start this Framework.
+     *
+     * <p>
+     * Calling this method is the same as calling {@link #Start()}. There are no
+     * start options for the Framework.
+     *
+     * @param options Ignored. There are no start options for the Framework.
+     * @throws std::runtime_error If this Framework could not be started.
+     * @see #Start()
+     */
+    void Start(uint32_t options);
```

Comment:
Simply for the sake of conciseness and maintenance, ```void Start(uint32_t options);``` could be modified to take a default value of ```0``` and ```void Start();``` could then be eliminated completely. This could also be applied to the base class ```Bundle```.